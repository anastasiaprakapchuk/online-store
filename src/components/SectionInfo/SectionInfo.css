.sectionBanners{
    position: relative;
}

.bannerOne{
    position: static;
}

.bannerOne img{
    width: 100%;
    position: relative;
    display: block;
}

.banner{
    position: absolute;
    transform: translateY(-100%);
    transition-duration: 2s;
}

.banner img{
    width: 100%;
    position: relative;
    display: block;
}

.sectionBanners .arrowHidden{
    position: absolute;
    left:0%;
    right: 0%;
    top:50%;
    transform: translateY(-50%); 
    display: flex;
    align-items: center;
    justify-content: space-between;
    opacity: 0;
    transition-duration: 1s;
}

.sectionBanners .arrowAppear{
    position: absolute;
    left:6%;
    right: 6%;
    top: 50%;
    transform: translateY(-50%); 
    display: flex;
    align-items: center;
    justify-content: space-between;
    opacity: 1;
    transition-duration: 1s;   
}

.sectionBanners .arrow{
    padding: 14px 19.5px;
    border-radius: 50%;
    background-color: white;
    opacity: 0.7;
    display: inline;
}

.sectionBanners .arrow:hover{
    cursor: pointer;
}





