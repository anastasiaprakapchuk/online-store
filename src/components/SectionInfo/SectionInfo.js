import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import { faAngleLeft,faAngleRight } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

import './SectionInfo.css';

export class SectionInfo extends React.PureComponent {
  static propTypes ={
    imagesInfo:PropTypes.object.isRequired,//получено от родителя
  };

  state = {
    appearBanner:0,
  };

  arrow=null;
 
  setArrow = (ref)=>{
    this.arrow=ref;
  }

  hoverArrow = ()=>{
    if(this.arrow)
    this.arrow.className='arrowAppear';
  }

  offHoverArrow = ()=>{
    if(this.arrow)
    this.arrow.className='arrowHidden';
  } 

  clickLeftArrow = () =>{
    let newBanner=this.state.appearBanner;
    if(this.state.appearBanner!==0)
    newBanner-=1;
    else
    newBanner=this.props.imagesInfo.data.length-1;
    this.setState({
      appearBanner:newBanner,
    });
  }

  clickRightArrow = () =>{
    let newBanner=this.state.appearBanner;
    if(this.state.appearBanner!==(this.props.imagesInfo.data.length-1))
    newBanner+=1;
    else
    newBanner=0;
    this.setState({
      appearBanner:newBanner,
    });
  }


  render (){
    //console.log(this.props.imagesInfo);
    let banners=this.props.imagesInfo.data.map((v,i)=>
    <div className={i===0?'bannerOne':'banner'} style={(i===this.state.appearBanner)?{opacity:'1'}:{opacity:'0'}} 
    key={i}><img src={v} alt=''></img></div>
    );
    return (
      <div className='sectionBanners' onMouseEnter={this.hoverArrow} onMouseLeave={this.offHoverArrow}>
        
        {banners}

        <div className='arrowHidden' ref={this.setArrow}>
           <div className="arrow arrowLeft" onClick={this.clickLeftArrow}> <FontAwesomeIcon icon={faAngleLeft}/> </div>
           <div className="arrow arrowRight" onClick={this.clickRightArrow}> <FontAwesomeIcon icon={faAngleRight}/> </div>
        </div>
        
      </div>
        
    );
  }
  }
  const mapStateToProps = function (state) {
    return {   
      imagesInfo: state.imagesInfo,      
    };
  };

export default connect(mapStateToProps)(SectionInfo);  