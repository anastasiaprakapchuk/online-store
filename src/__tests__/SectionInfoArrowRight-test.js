"use strict";

import React from 'react';
import { SectionInfo } from '../components/SectionInfo/SectionInfo';
import renderer from 'react-test-renderer';


let imagesInfo={data:["imagesInfo/wish.jpg","imagesInfo/sale.jpg","imagesInfo/sale8m.jpg"] };
         

test('работа стрелки "Вправа"', () => {

  // создаём тестовую версию компонента
  const component = renderer.create(
    <SectionInfo imagesInfo={imagesInfo}/>
  );

  // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  // найдём в вёрстке компонента саму стрелку "Вправа"
  const buttonElem = component.root.find( el => el.props.className=="arrow arrowRight"); 
  console.log(buttonElem);
  // и "нажмём" на неё
  buttonElem.props.onClick();

  // получаем уже изменённый снэпшот
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  });

  