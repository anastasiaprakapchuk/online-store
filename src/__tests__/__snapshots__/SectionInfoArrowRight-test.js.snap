// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа стрелки "Вправа" 1`] = `
<div
  className="sectionBanners"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="bannerOne"
    style={
      Object {
        "opacity": "1",
      }
    }
  >
    <img
      alt=""
      src="imagesInfo/wish.jpg"
    />
  </div>
  <div
    className="banner"
    style={
      Object {
        "opacity": "0",
      }
    }
  >
    <img
      alt=""
      src="imagesInfo/sale.jpg"
    />
  </div>
  <div
    className="banner"
    style={
      Object {
        "opacity": "0",
      }
    }
  >
    <img
      alt=""
      src="imagesInfo/sale8m.jpg"
    />
  </div>
  <div
    className="arrowHidden"
  >
    <div
      className="arrow arrowLeft"
      onClick={[Function]}
    >
       
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-left fa-w-8 "
        data-icon="angle-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
       
    </div>
    <div
      className="arrow arrowRight"
      onClick={[Function]}
    >
       
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-right fa-w-8 "
        data-icon="angle-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
       
    </div>
  </div>
</div>
`;

exports[`работа стрелки "Вправа" 2`] = `
<div
  className="sectionBanners"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="bannerOne"
    style={
      Object {
        "opacity": "0",
      }
    }
  >
    <img
      alt=""
      src="imagesInfo/wish.jpg"
    />
  </div>
  <div
    className="banner"
    style={
      Object {
        "opacity": "1",
      }
    }
  >
    <img
      alt=""
      src="imagesInfo/sale.jpg"
    />
  </div>
  <div
    className="banner"
    style={
      Object {
        "opacity": "0",
      }
    }
  >
    <img
      alt=""
      src="imagesInfo/sale8m.jpg"
    />
  </div>
  <div
    className="arrowHidden"
  >
    <div
      className="arrow arrowLeft"
      onClick={[Function]}
    >
       
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-left fa-w-8 "
        data-icon="angle-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
       
    </div>
    <div
      className="arrow arrowRight"
      onClick={[Function]}
    >
       
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-right fa-w-8 "
        data-icon="angle-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
       
    </div>
  </div>
</div>
`;
